### ISSUE TITLE
Master is failing to build due to issues in crash_reporter.cc

### ISSUE THREAD

**Issue by frankhale** [atom/electron#1910]
I just got a fresh clone of Electron and I'm trying to build on Windows but I'm seeing issues with crash_reporter.cc which are causing the build to fail.

```
C:\Users\FrankDev\Desktop\build\electron>python script\build.py -c D
ninja: Entering directory `out\D'
[323/1060] CC obj\vendor\node\deps\openssl\openssl\crypto\dso\openssl.dso_win32.obj
c:\users\frankdev\desktop\build\electron\vendor\node\deps\openssl\openssl\crypto\dso\dso_win32.c(289) : warning C4055: 'type cast' : from data
 pointer 'void *' to function pointer 'DSO_FUNC_TYPE'
[739/1060] CC obj\vendor\node\deps\openssl\openssl\crypto\whrlpool\openssl.wp_block.obj
c:\users\frankdev\desktop\build\electron\vendor\node\deps\openssl\openssl\crypto\whrlpool\wp_block.c(87) : warning C4164: '_rotl64' : intrinsi
c function not declared
[841/1060] CXX obj\atom\common\crash_reporter\electron_lib.crash_reporter.obj
FAILED: ninja -t msvc -e environment.x64 -- "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64\cl.exe" /nologo /showIncludes /F
C @obj\atom\common\crash_reporter\electron_lib.crash_reporter.obj.rsp /c ..\..\atom\common\crash_reporter\crash_reporter.cc /Foobj\atom\common
\crash_reporter\electron_lib.crash_reporter.obj /Fdobj\electron_lib.cc.pdb
c:\users\frankdev\desktop\build\electron\atom\common\crash_reporter\crash_reporter.cc(49) : error C2440: '<function-style-cast>' : cannot conv
ert from 'const std::string' to 'base::FilePath'
        No constructor could take the source type, or constructor overload resolution was ambiguous
[841/1060] CXX obj\atom\utility\electron_lib.atom_content_utility_client.obj
ninja: build stopped: subcommand failed.
```

**Comment by frankhale** [comment]
Possible related patch: https://github.com/atom/electron/commit/4c48f0cd3698c617c6c25c44d203dc0d41f66c14

**Comment by frankhale** [comment]
I naively made the following change in order to convert the std::string path to a FilePath and ultimately to get master to compile. 

I changed:

```cpp
base::FilePath(path)
```

to

```cpp
base::FilePath::FromUTF8Unsafe(path)
```

Here is the full function in crash_reporter.cc

```cpp
std::vector<CrashReporter::UploadReportResult>
CrashReporter::GetUploadedReports(const std::string& path) {
  std::string file_content;
  std::vector<CrashReporter::UploadReportResult> result;

  if (base::ReadFileToString(base::FilePath::FromUTF8Unsafe(path), &file_content)) {
    std::vector<std::string> reports;
    base::SplitString(file_content, '\n', &reports);
    for (const std::string& report : reports) {
      std::vector<std::string> report_item;
      base::SplitString(report, ',', &report_item);
      int report_time = 0;
      if (report_item.size() >= 2 && base::StringToInt(report_item[0],
            &report_time)) {
        result.push_back(CrashReporter::UploadReportResult(report_time,
            report_item[1]));
      }
    }
  }
  return result;
}
```

**Comment by kirkouimet** [comment]
This worked for me, I still am having trouble compiling on Windows, but I put this change into a pull request here:

https://github.com/atom/electron/pull/1918

**Comment by frankhale** [comment]
Thanks for putting in a pull request!

**Comment by frankhale** [comment]
@kirkouimet, I was also having issues compiling on Windows recently and started to have to set an environment variable to make it work. I use the Visual Studio Command Prompt to compile and before I compile I set this: 

```
set GYP_MSVS_VERSION=2013
```

If I don't set this the compilation will just bomb out at random times. Once you set that then do your  bootstrap and build. Hopefully it will resolve your compile issues.

**Comment by zcbenz** [comment]
Fixed by #1918.

**Comment by frankhale** [comment]
Thanks @zcbenz & @kirkouimet
